org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DOMToModelPopulator.java

```diff
@@ -701,4 +732,47 @@ private ServiceInfo toServiceInfo(ProvidesDirective node) {
 		res.setSourceRangeEnd(node.getStartPosition() + node.getLength() - 1);
 		return res;
 	}
+	private boolean isNodeDeprecated(BodyDeclaration node) {
+		return ((List<ASTNode>)node.modifiers()).stream() //
+				.anyMatch(modifier -> {
+					if (!isAnnotation(modifier)) {
+						return false;
+					}
+					String potentiallyUnqualifiedAnnotationType = ((org.eclipse.jdt.core.dom.Annotation)modifier).getTypeName().toString();
+					if ("java.lang.Deprecated".equals(potentiallyUnqualifiedAnnotationType)) { //$NON-NLS-1$
+						return true;
+					}
+					return "Deprecated".equals(potentiallyUnqualifiedAnnotationType) && !hasAlternativeDeprecated(); //$NON-NLS-1$
+				});
+	}
+	private static boolean isAnnotation(ASTNode node) {
+		int nodeType = node.getNodeType();
+		return nodeType == ASTNode.MARKER_ANNOTATION || nodeType == ASTNode.SINGLE_MEMBER_ANNOTATION
+				|| nodeType == ASTNode.NORMAL_ANNOTATION;
+	}
+	private boolean hasAlternativeDeprecated() {
+		if (this.alternativeDeprecated != null) {
+			return this.alternativeDeprecated;
+		}
+		try {
+			IJavaElement[] importElements = this.importContainer.getChildren();
```

Comment:
This has introduced a NPE which causes many tests to fail later on.
Before this patch, the log were showing "Tests run: 22846, Failures: 1553, Errors: 142, Skipped: 0"; now it's "Tests run: 22846, Failures: 1576, Errors: 2000, Skipped: 0" (there might be other regressions involved).
I will try to get more test reports shown on PRs.